cg/apps/demultiplex/sample_sheet/read_sample_sheet.py

```diff
@@ -110,14 +110,15 @@ def get_samples_by_lane(
 
 
 def get_sample_internal_ids_from_sample_sheet(
-    sample_sheet_path: Path, flow_cell_type: Type[FlowCellSample]
+    sample_sheet_path: Path, flow_cell_sample_type: Type[FlowCellSample]
```

Comment:
What does `Type` do?

Reply:
Do we need to pass `flow_cell_sample_type`? Is it not more logical to be able to just give a sample_sheet and have the function figure out what type it is?

Reply:
The sample sheet only hold information on the samples and not the sequencer or sequencer type. I think from that alone it will be difficult to know what flow cell sample type is used. I tried to conform to how samples are parsed upstream when reading sample sheets in this module, that is why the usage of the flow cell sample type is there.

Do you think it an issue to have it like this?

Reply:
So the sample sheets differ between types? If so the information is implicitly there right?

Reply:
I think the other way would be to have a check for column naming in the sample sheet which would be similar to what to the logic from the flow cell sample type. I will open a new PR that opens this possibility and then we can decide there.